feat: send notification to authors

orlera opened the pull request:
#### :tophat: What? Why?
Send's a notification to the content's author when their content gets reported.

#### Testing
 - Login with a user (say: user@example.org | decidim123456);
 - Create a reportable content (e.g. a comment, proposal, etc.);
 - On another browser/session, login with another user (say: admin@example.org | decidim123456);
 - Report the content created with the previous user;
 - In the first browser/session, you shell receive a notification saying that you content has been reported;

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] :question: **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] :heavy_check_mark: **DO** check open PR's to avoid duplicates.
- [ ] :heavy_check_mark: **DO** keep pull requests small so they can be easily reviewed.
- [ ] :heavy_check_mark: **DO** build locally before pushing.
- [ ] :heavy_check_mark: **DO** make sure tests pass.
- [ ] :heavy_check_mark: **DO** make sure any new changes are documented in `docs/`.
- [ ] :heavy_check_mark: **DO** add and modify seeds if necessary.
- [ ] :heavy_check_mark: **DO** add CHANGELOG upgrade notes if required.
- [ ] :heavy_check_mark: **DO** add to GraphQL API if there are new public fields.
- [ ] :heavy_check_mark: **DO** add link to MetaDecidim if it's a new feature.
- [ ] :x:**AVOID** breaking the continuous integration build.
- [ ] :x:**AVOID** making significant changes to the overall architecture.

### :camera: Screenshots
![notification](https://user-images.githubusercontent.com/5033945/95986717-b7cacf00-0e26-11eb-8cc3-6533dbb416b8.gif)


:hearts: Thank you!

roxanaopr committed:
Ref. DIFE-148
- add registration_code_enabled as an admin flag on meetings settings page
- remove registration code from email and notification
- do not display registration code on the interface after joining to a meeting

roxanaopr committed:
Ref. DIFE-148
- hide "Validate registration code" from Admin - Components - Meeting section

roxanaopr committed:
Ref. DIFE-148
- fix rspec error

roxanaopr committed:
Merge branch 'develop' into feature/disable-registration-code

roxanaopr committed:
Ref. DIFE-148
- fix bad reference

roxanaopr committed:
Ref. DIFE-148
- fix tests for join_meeting_spec.rb

roxanaopr committed:
Ref. DIFE-148
- fix tests for meeting_registrations_spec.rb

roxanaopr committed:
Ref. DIFE-148
- fix tests for registration_mailer_spec.rb

roxanaopr committed:
Ref. DIFE-148
- fix tests for admin_manages_meetings_spec.rb

roxanaopr committed:
Ref. DIFE-148
- fix tests for validate_registration_code_spec.rb

roxanaopr committed:
Ref. DIFE-148
- fix linter errors

alecslupu committed:
Merge branch 'develop' into feature/disable-registration-code

alecslupu committed:
Adding Configuration Settings for Demographic data

alecslupu committed:
Adding the Demographics module

nocode66 committed:
made form persistent

alecslupu committed:
Refactoring demographic module to enforce decidim standard

alecslupu committed:
improving demographic engine codebase

alecslupu committed:
fixing error caused by Missing Organization

alecslupu committed:
Merge branch 'fix/user-exporter-job' into feature/add-demographic-module

alecslupu committed:
fixing norvegian file

alecslupu committed:
Merge branch 'fix-internationalization' into feature/add-demographic-module

alecslupu committed:
Merge branch 'develop' into feature/disable-registration-code

alecslupu committed:
Merge branch 'develop' into feature/add-demographic-module

alecslupu committed:
Merge branch 'develop' into feature/disable-registration-code

alecslupu committed:
Cleanup

nocode66 committed:
graphql attempt

alecslupu committed:
Fixing typos

alecslupu committed:
Adding Suspend User mechanism

alecslupu committed:
Linting and remove Geocoding raise issue

alecslupu committed:
Linting and remove Geocoding raise issue

alecslupu committed:
Normalize locales

alecslupu committed:
Normalize locales

beagleknight committed:
feat: add extra columns to the moderation panel

beagleknight committed:
feat: sort moderations by report count and creation date

beagleknight committed:
feat: filter moderations by reportable type

roxanaopr committed:
Notify user via email in case that his account was banned/blocked

roxanaopr committed:
Update user_suspension text and rearrange blocks in en.yml

alecslupu committed:
Merge pull request #7 from decidim/develop

Bump develop to next release version

beagleknight committed:
chore: add reported_content to moderations migration

nocode66 committed:
added graphql demographic

nocode66 committed:
removed unused translation

nocode66 committed:
removed whitespace

beagleknight committed:
feat: filter moderations by reportable id or reported content

beagleknight committed:
feat: update reported content when a report is created

beagleknight committed:
feat: add pagination links

beagleknight committed:
test: prevent random failures not using random numbers

beagleknight committed:
chore: fix some minor problems

ivan-mr committed:
Fix deprecated js loadMap on meetings index (#6654)

* Fix deprecated js loadMap on meetings index

* Fix lint error index.js.erb removing whitespace at end of line

* Fix same lint error on meetings index.js.erb whitespace at end of line

beagleknight committed:
Merge pull request #12 from tremend-cofe/feat/sort-moderation-panel

mrcasals committed:
Merge pull request #13 from tremend-cofe/feat/filter-moderation-panel

alecslupu committed:
Merge branch 'develop' of github.com:tremend-cofe/decidim into feature/add-demographic-module

beagleknight committed:
Merge pull request #11 from tremend-cofe/feat/moderation-panel-improvements

alecslupu committed:
Linters

alecslupu committed:
Refactory on demographic module

alecslupu committed:
Linters

decidim-bot committed:
New Crowdin updates (#6619)

* New translations en.yml (Catalan)

* New translations en.yml (Catalan)

* New translations en.yml (Spanish, Paraguay)

* New translations en.yml (French, Canada)

* New translations en.yml (French, Canada)

* New translations en.yml (French, Canada)

* New translations en.yml (Norwegian)

* New translations en.yml (Dutch)

* New translations en.yml (Dutch)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Spanish)

* New translations en.yml (Catalan)

* New translations en.yml (Spanish, Mexico)

* New translations en.yml (Spanish, Paraguay)

* New translations en.yml (Spanish)

* New translations en.yml (Catalan)

* New translations en.yml (Spanish, Paraguay)

* New translations en.yml (Spanish, Mexico)

* New translations en.yml (Norwegian)

* New translations en.yml (Korean)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Somali)

* New translations en.yml (Oromo)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Somali)

* New translations en.yml (Oromo)

* New translations en.yml (Amharic)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Tigrinya)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Oromo)

* New translations en.yml (Amharic)

* New translations en.yml (Amharic)

* New translations en.yml (Korean)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Korean)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Korean)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Vietnamese)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Chinese Traditional)

* New translations en.yml (Tigrinya)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (Chinese Simplified)

* New translations en.yml (French)

* New translations en.yml (Bulgarian)

* New translations en.yml (Catalan)

* New translations en.yml (Czech)

* New translations en.yml (Bulgarian)

* New translations en.yml (Bulgarian)

* New translations en.yml (Bulgarian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Italian)

* New translations en.yml (Finnish)

* New translations en.yml (Finnish)

* New translations en.yml (Finnish)

* Rebuild JS bundle

* New translations en.yml (Dutch)

* Rebuild JS bundle

Co-authored-by: Ramon Costa <ramon.costa@gmail.com>

slickepinne committed:
Fix order for elections (#6616)

* fix order for elections

* remove unneeded css class from test

Commit:
wip create and trigger report_created event

Commit:
linting

roxanaopr committed:
Update Admin Loog to include justification for "suspend" action

oriolgual commented on decidim-core/app/commands/decidim/create_report.rb:
```diff
@@ -70,6 +71,16 @@ def send_report_notification_to_moderators
+      data = {
+        event: "decidim.events.reports.report_created",
+        event_class: Decidim::ReportCreatedEvent,
+        resource: @report,
```
Maybe the resource here should be `@report.moderation.reportable`?

oriolgual commented on decidim-core/app/events/decidim/report_created_event.rb:
```diff
@@ -0,0 +1,28 @@
+# frozen-string_literal: true
+
+module Decidim
+  class ReportCreatedEvent < Decidim::Events::SimpleEvent
```
You probably want to include [AuthorEvent](https://github.com/decidim/decidim/blob/develop/decidim-core/lib/decidim/events/author_event.rb)

orlera replied:
Hm, I'm not sure about this one, because at the moment we're not really using any info from the author in `ReportCreatedEvent`.

oriolgual commented on decidim-core/app/events/decidim/report_created_event.rb:
```diff
@@ -0,0 +1,28 @@
+      "localhost:3000/reported_resource"
+    end
+
+    def resource_title
```
The default `resource_title` might be enough, it tries to get a meaningful title: https://github.com/decidim/decidim/blob/develop/decidim-core/lib/decidim/events/base_event.rb#L72-L82

roxanaopr committed:
Modify user name to "Blocked user" when his account is blocked

agustibr committed:
Delete decidim-participatory_processes/A file (#6668)

alecslupu committed:
fixing error caused by Missing Organization (#6612)

andreslucena committed:
Remove unused /organization_users route (#6673)

armandfardeau committed:
Homepage content blocks cache (#6235)

* Add cache to hero cell

* Add cache to last activity

* Fix rubocop offenses

* Don't herit perform caching method

* add current locale to cache hash

* add current locale to cache hash

* Fix offenses

* Add comment on method cache_hash

* Revert back to original locale after test

* Use a regex instead of changing locale

* Fix offense

andreslucena committed:
Remove unused Initiative author i18n block (#6667)

segoy89 committed:
Fix nickname prefix wraps on certain view widths (#6649)

andreslucena committed:
Fix error when saving an Initiative title and description from Admin (#6581)

alecslupu committed:
Adding tests

andreslucena committed:
Fix Initiative's neutral vocabulary (#6590)

Changes this vocabulary specific for a city/municipal context to
something much more neutral.

Co-authored-by: Oliver Valls <199462+tramuntanal@users.noreply.github.com>

leio10 committed:
fix: avoid rendering values on the secrets.yml when creating a new app (#6653)

It should access to the code at runtime, and access the environment variables.

andreslucena committed:
Fix default help section and page for Conferences (#6618)

* Add help section and page to Conferences

* Fix plural/singular in Consultations' help

* Fix i18n normalize

Commit:
Add scope to proposals import (#6525)

* Add scope filter to proposals import

* Update specs

* Fix offense

* Change approach: scope selector filtered by current budget's scope

* Remove unused key in locales

* Use symbol instead of string

* Add test

* Fix offense

* Render scope picker only when scopes are enabled

* Rename decidim_scope_id to scope_id

* Rename remaining decidim_scope_id vars to scope_id

* Fix test

Commit:
improve notification message

andreslucena committed:
Add new languages: Korean, Vietnamese and Chinese (Traditional and Simplified) (#6648)

Commit:
use reportable as event resource

mllocs committed:
Implement close meeting in public pages

Commit:
retrieve authors

Commit:
add test

andreslucena committed:
Remove show_statistics checkbox in Appearance (#6575)

* Remove show_statistics checkbox in Admin -> Settings -> Appearance

* Fix 'Missing magic comment' lint error

* Restore Homepage system spec

* Fix failing tests for Homepage system spec

* Add CHANGELOG entry with removed option

* Fix lint error in CHANGELOG.md

mllocs committed:
Added specs

anaghavl committed:
Add online meetings (#6572)

* Add config to allow online meetings

* Add online_meeting_url fields to DB

* Add fields to forms and commands

* Use `data-` attributes instead of specific field names

* Remove unused locales

* Fix JS

* Fix locale

* Fix forms specs

* Fix specs

* Hide meeting type and online url fields if online meetings are not allowed

* Show online meeting URL

* Ensure fields are hidden

* Fix meeting form

* Fix tests

* Add `type_of`meeting` column to meetings

It stores whether the meeting is in-person or online

* Filter meetings by type

* Add missing i18n string

* Validate URLs

* Address feedback

* Fix i18n

* Remove config option

* test: select the type of meeting before filling in location

* Move URL details to cell

* Fix form

* Fix icon

* Fix edit form

Co-authored-by: Marc Riera Casals <mrc2407@gmail.com>
Co-authored-by: David Morcillo <david.morcillo@gmail.com>

alecslupu committed:
Display the time Rails spends rendering a cell (#6515)

* Adding cell instrumentation to display rails-cells information output

* Linters

* Refactory of the Instrumenter system, moved into Decidim ViewModel

roxanaopr committed:
Render an inaccessible user profile page for blocked accounts

Commit:
add link to participatory space

Commit:
add reported content id

roxanaopr committed:
Add missing file for rendering an inaccessible user profile page

Commit:
linting

Commit:
remove unused eky

SilviaGrrr committed:
Add missing margin between username and title in cards (#6674)

* First try -fix issue #6545 (bottom margin card title); not sure it works.

* Second try -fix issue #6545 (bottom margin card title); still not sure it works

* Fix issue #6545 (bottom margin card title), 2n ed. femdev

* Fix issue #6545 (bottom margin card title), 2n ed. femdev

* Added missing margin between card title and author, issue #6545; includes styling conventions required by codeclimate automated test.

beagleknight commented on decidim-core/app/commands/decidim/create_report.rb:
```diff
@@ -70,6 +71,19 @@ def send_report_notification_to_moderators
     end
 
+    def send_report_notification_to_author
+      data = {
+        event: "decidim.events.reports.report_created",
+        event_class: Decidim::ReportCreatedEvent,
+        resource: @report.moderation.reportable,
+        extra: {
+          report_reason: @report.reason
+        },
+        affected_users: @report.moderation.reportable.try(:authors) || [@report.moderation.reportable.try(:author)]
+      }
+      Decidim::EventsManager.publish(data)
```
I may lack of context but I feel like building this data structure should be in the `Decidim::ReportCreatedEvent` class, right? 🤔 
Something like:

```rb
event = Decidim::ReportCreatedEvent.new(@report)
Decidim::EventsManager.publish(event.to_data)
```

orlera replied:
That'd make a lot of sense. Still, all the events I've checked out are passed to the `EventManager` this way (the caller builds the event data), so I'm not sure we want to introduce a new way of doing it here.
What are your thoughts on this @mrcasals?

mrcasals replied:
You are... right. This makes more sense, but the current approach is the one Andrea used...

beagleknight reviewed (approved):
LGTM! 👍 I just added a small comment but I may be wrong because I don't have a lot of context of this part.

roxanaopr committed:
Remove duplicate call of notify_uer method

ahukkanen committed:
Fix the file validator humanizer with static numeric values (#6682)

ahukkanen committed:
Allow customization of the upload help messages (#6683)

andreslucena committed:
Fix elections count in Homepage statistics (#6684)

roxanaopr committed:
Fix some linting erors.

Commit:
add content original language

mrcasals committed:
Ensure `resource_text` is a string in NotificationMailer (#6685)

* Ensure strings in NotificationMailer

* Add specs

* Ensure all events return a string on `safe_resource_text`

ovruni committed:
Fix pending authorization list (#6680)

alecslupu committed:
Merge branch 'develop' of github.com:decidim/decidim into develop

alecslupu committed:
Merge branch 'develop' into fix-foe

Commit:
test id presence

roxanaopr committed:
Normalize locales

Commit:
fix test

mllocs committed:
Added system spec

alecslupu committed:
Adding functionality to report users

mllocs committed:
Update decidim-meetings/app/commands/decidim/meetings/close_meeting.rb

Co-authored-by: Oriol Gual <oriolgual@users.noreply.github.com>

mllocs committed:
Fixed closing_report edit field when not empty

mllocs committed:
Fixed lint error

mllocs committed:
Merge branch 'fix-foe' into feature/close-meeting-in-public-pages

alecslupu committed:
Adding admin log for unreport, i18n string fixes

alecslupu committed:
Running linters

alecslupu committed:
Normalize locales

alecslupu committed:
Adding UserGroup Compatibility, Adding Tests

alecslupu committed:
Adding tests for reported user functionality

alecslupu committed:
Running linters

alecslupu committed:
Fixing failing test, Running linters

alecslupu committed:
Merge branch 'develop' into feature/add-demographic-module

ahukkanen committed:
Refactor comments: get rid of react (#6498)

* Move decidim comments to cells instead of react

* Move the comment form to its own cell and add reply form

* Fix JS errors in the comments component

* Add backend functionality for answering a comment

* Updates to the comments cells

- Fix display of comment replies and the conversation title
- Show reply only when able to reply
- Add blocked comments messages

* Fix create comment action

* Implement comment form dynamic actions

- Add functionality to the front-end comment alignment buttons
- Implement enabling/disabling the send comment button depending
  on the comment field content

* Mount the comments engine

* Updates with the cells and dynamic functionality

- Implement add comment and reply to comment
- Fix problems with the cells in different situations and the
  single comment view

* Implement polling for new comments

* Implement comment votes

* Implement reorder of comments

* Implement the comment as dropdown

* Use the SortedComments query for the comments polling

This makes the comments polling order aware.

* Normalize the locales

* npm run build:prod

* Remove the comments JS bundle check

* ERBLint

* Hide comment votes when they are disabled

* Format comment body correctly

* Fix the manual non-db sortings for comments

* Allow the comments index action for non-authorized users

* Add comment mentions possibility

* Fix the comments helper spec

* Fix comments examples

* Fix organization comments max length

* Ingore the comments index.js from ERBLint

* Fix commentable path for budgeting projects

* Implement comment alignment badge

* Fix ERBLint issue with comments index.js.erb

* Implement comment flagging

* Improve the comments front-end component

- Move the listeners to their own methods
- Fix some typos

* Fix tribute warnings

* Fix duplicate IDs in the DOM

* Add tests for the vanilla JS controlling dynamic comment functionality

* Fix issue with the conflicting cell name after the merge

* Fix crash after the merge

Due to the duplicate cell definitions, there was a crash in the
system that caused startup to fail.

* Fix order link click handler

* Remove unnecessary double if statement

* Only display flag modal if the user is signed in

* Return no groups if the user is not signed in

* Fix comments section title with only 1 comment

* Add specs for the cells and the controllers

* Add CHANGELOG entry

* Fix issue with finding hidden elements

* Fix legacy element selector in the comments examples

* Remove comment body label

* Mitigate issue when the component does not define comments_max_length

* Fix spec names

* Do not show or allow creation of comments when they are disabled

* Disallow the user from commenting a private space commentable

* Add commentable checks to the dummy resource

- Add `commentable?` method to check that the component comments
  are enabled as with most other objects
- Add `user_allowed_to_comment?` method to check that the user
  can participate in the space as with most other objects

* Do not allow to comment if the user is not defined

* Add comments permissions spec

* Move the with_comments_disabled to a trait in the component factory

* Disallow commenting if the comments are not enabled

* Check the commentable/comments enabled through permissions

* Fix the create comment spec when comments are disabled

* Move the permission class chain to the comments application controller

* Add permission checks for the comments

* Make comment_maximum_length more robust

The comment length comparison would fail if this returns nil.

* Add dynamic update of the comments count

* Fix the permissions spec failing with other tests

* Remove 0.23 note from the CHANGELOG

mrcasals committed:
Add machine translations on factories (#6665)

* Add machine translations on factories

* Fix system specs

* Handle machine translations on test helper

* Fix proposal unit tests

* Fix sortitions specs

* Fix proposal diff renderer

* Fix process specs

* Fix blogs specs

* Fix initiatives specs

* Fix debates specs

* Fix core specs

* Fix conferences specs

* Fix conferences specs

* Fix budgets tests

* Fix assembly specs

* Fix accountability specs

* Fix admin specs

* Fix blogs spec

* Fix magic comment

* Lint file

mllocs committed:
Added close_meeting command specs

mllocs committed:
Added close meeting form specs

mllocs committed:
Removed empty lines

mllocs committed:
Fixed Linting error

alecslupu committed:
Partial commit with demographics

alecslupu committed:
Merge branch 'develop' of github.com:decidim/decidim into fix-foe

alecslupu committed:
Merge branch 'develop' of github.com:decidim/decidim into develop

alecslupu committed:
Merge branch 'develop' into fix-foe

mrcasals committed:
Merge pull request #16 from tremend-cofe/feature/close-meeting-in-public-pages

alecslupu committed:
Merge branch 'feature/add-demographic-module' into fix-foe

alecslupu committed:
Merge branch 'feature/add-report-user-functionality' into fix-foe

alecslupu committed:
Merge branch 'feature/add-suspend-user-functionality' into fix-foe

alecslupu committed:
Merge branch 'feature/disable-registration-code' into fix-foe

alecslupu committed:
Merge branch 'fix-foe' of github.com:tremend-cofe/decidim into fix-foe

alecslupu committed:
Running linters on tests

alecslupu committed:
Merge branch 'feature/add-demographic-module' into fix-foe

roxanaopr committed:
Normalize locales en

roxanaopr committed:
Fix linting error in erb file

alecslupu committed:
Merge branch 'develop' of github.com:tremend-cofe/decidim into feature/add-suspend-user-functionality

alecslupu committed:
Merge branch 'feature/add-suspend-user-functionality' of github.com:tremend-cofe/decidim into feature/add-suspend-user-functionality

alecslupu committed:
Linting erb

alecslupu committed:
Merge branch 'feature/add-suspend-user-functionality' into fix-foe

alecslupu committed:
Linters

alecslupu committed:
Merge branch 'feature/add-suspend-user-functionality' into fix-foe

tramuntanal committed:
Refactor meetings test to be resilient to flakys (#6694)

* Refactor test to be resilient to flakys

* Wait longer for the ajax to finish

* Add temporal logs for debugging CI

* More traces

* Remove wait time as it doesn't seem to be related with the problem

* More traces

* Getting closer...

* Remove useless lines

* Try to make budgets test more reliable

* Remove traces

* Remove `puts` in view

alecslupu committed:
Merge branch 'develop' of github.com:decidim/decidim into fix-foe

mrcasals reviewed (approved):
Looks good to me!

orlera committed:
Merge pull request #17 from tremend-cofe/feat/add-info-to-moderator-email

feat: add info to moderator email

Commit:
Merge branch 'fix-foe' into feat/send-notification-to-authors
